engine/src/main/java/org/hibernate/search/engine/search/predicate/dsl/SimpleQueryFlagsStep.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.search.predicate.dsl;
+
+import java.util.EnumSet;
+
+/**
+ * The step in a flags definition where the simple query can be set.
+ *
+ * @param <S> The "self" type (the actual exposed type of this step)
+ */
+public interface SimpleQueryFlagsStep<S extends SimpleQueryStringPredicateOptionsStep<?>> extends SimpleQueryStringPredicateOptionsStep<S> {
```

Comment:
If you are going to make this step extend `OptionsStep`, I think you can simply remove the `flags` method and move all these methods to `OptionsStep`.